### ISSUE TITLE
Fix theme selection

### ISSUE THREAD

**Issue by Beep6581** [Beep6581/RawTherapee#2980]
RT by default loads the bundled Adwaita CSS file (gtk-contained-dark.css) and then loads the RawTherapee.css file on top of that. This is to keep RawTherapee looking consistent even if a Gtk3 user does not have the Adwaita theme, or uses a different theme system-wide. We need to decide whether the bundled Adwaita CSS file should also be loaded first if the user makes and selects their own theme, or whether that own CSS should be the only one loaded. Doing the former leads to a predictable starting point (always Adwaita) while the latter will lead to an unpredictable starting point, based upon the user's Gtk3 system-wide theme if its not Adwaita. This means that if the user intends to make a new high quality CSS theme and have us bundle it and the user's system theme was not Adwaita and we went with the latter, then their new high quality CSS theme would look broken on everyone else's system assuming everyone else doesn't use the same system-wide theme as the designer did.
Furthermore, the chance of someone making a new, high quality CSS theme which we would want to bundle is proportionally low to the priority of this issue ;]

**Comment by Beep6581** [comment]
RawTherapee no longer bundles or loads `gtk-contained-dark.css`, now it invokes the dark Adwaita variant using Gtk+:
```
        Gtk::Settings::get_for_screen(screen)->property_gtk_theme_name() = "Adwaita";
        Gtk::Settings::get_for_screen(screen)->property_gtk_application_prefer_dark_theme() = true;
```
Then it applies the user-chosen theme, which is currently hard-coded to `RawTherapee.css`.

**Comment by Beep6581** [comment]
@TooWaBoo should we call your theme "TooWaBlue"?

**Comment by Beep6581** [comment]
Pull request opened:
https://github.com/Beep6581/RawTherapee/pull/3202

**Comment by TooWaBoo** [comment]
@Beep6581 
You can call the theme whatever you want. Maybe call it RawTherapeeBlue. It's up to you.
For the new version 4.2.849, I have to rework the whole theme again. There many changes to the object refference.
I'll hope I can finish it until weekend.

**Comment by Hombre57** [comment]
@TooWaBoo
Unfortunately, the main theme will change again in the future (in few months I guess) when Gtk 3.20 will be available. Colors shouldn't change anymore (well... maybe for the Preference window, if I'll still care), but margins and paddings should, since they are partly supported by Gtk3 as of now (3.18), i.e. the margin and paddings already set in rawtherapee.css for Gtk::Frame are unused by Gtk.

**Comment by TooWaBoo** [comment]
TooWaBlue for RT 4.2.849 finished :-)
Rename file extension from .txt to .css
[RawTherapee.txt](https://github.com/Beep6581/RawTherapee/files/163829/RawTherapee.txt)

**Comment by Beep6581** [comment]
Thanks, I'll test and include it in PR #3202 which I will merge tomorrow as no one has objected. We can update it afterwards as needed.

**Comment by TooWaBoo** [comment]
@Hombre57 
Does this mean, that margin is supported in the future?

**Comment by Hombre57** [comment]
That's what @Beep6581 was told by the Gtk dev : this is something they are implementing/enhancing for Gtk 3.20.

**Comment by Hombre57** [comment]
`cssBase` could be used to load the Adwaita theme (if we bundle it, for the reason you explained in the initial post), so I guess that you've renounced to that ? If so, I've no objection on committing.

**Comment by Beep6581** [comment]
While writing the first post I thought bundling Adwaita*.css was what we had to do, but then I learned of the better way which I used in the code and in the second post. So we still do load Adwaita first and then overwrite it with a custom theme. This makes sense is we want our theme to be similar to Adwaita and only differ in a few places, as it makes our custom .css shorter. But if we want our custom .css to differ a lot, then it might make more sense to skip loading Adwaita altogether, because the loading times might be shorter. The reason I say "might" is because I don't know what happens if we skip those two lines from the second comment - maybe Gtk+ will still load the Adwaita theme even without those lines, as the theme is central to Gtk+ 3, with the only difference being it would load the light variant, and so loading time won't change. I'll try to measure theme loading times.
